unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -551,21 +551,35 @@ abstract public class BaseFPSAgentController {
                 actionReturn: reachablePositions
             );
         }
+        
+        protected virtual void SetAgentMode() {
+            VisibilityCapsule = TallVisCap;
+            m_CharacterController.center = new Vector3(0, 0, 0);
+            m_CharacterController.radius = 0.2f;
+            m_CharacterController.height = 1.8f;
+
+            CapsuleCollider cc = this.GetComponent<CapsuleCollider>();
+            cc.center = m_CharacterController.center;
+            cc.radius = m_CharacterController.radius;
+            cc.height = m_CharacterController.height;
+
+            m_Camera.GetComponent<PostProcessVolume>().enabled = false;
+            m_Camera.GetComponent<PostProcessLayer>().enabled = false;
+
+            // camera position
+            m_Camera.transform.localPosition = new Vector3(0, 0.675f, 0);
+
+            // camera FOV
+            m_Camera.fieldOfView = 90f;
+
+            // set camera stand/crouch local positions for Tall mode
+            standingLocalCameraPosition = m_Camera.transform.localPosition;
+            crouchingLocalCameraPosition = m_Camera.transform.localPosition + new Vector3(0, -0.675f, 0); // bigger y offset if tall
+        }
 
         public void Initialize(ServerAction action) {
-            if (action.agentMode.ToLower() == "default" ||
-               action.agentMode.ToLower() == "locobot" ||
-               action.agentMode.ToLower() == "drone" ||
-               action.agentMode.ToLower() == "arm"
-               ) {
-                // set agent mode to Default, Bot or Drone accordingly
-                SetAgentMode(action.agentMode);
-            } else {
-                errorMessage = "agentMode must be set to 'default' or 'bot' or 'drone' or 'hand'";
-                Debug.Log(errorMessage);
-                actionFinished(false);
-                return;
-            }
+            this.SetAgentMode();
```

Comment:
I'm not sure the name `SetAgentMode` really makes sense anymore. Can we change this to `InitializeAgent` or something similar?

Reply:
I agree.  How does everyone feel about `initializeBody`? Pretty much all the code in `SetAgentMode()` has to do with setting up the capsule and possibly the arm.

Reply:
`InitializeBody` sounds good to me!